lib/groupio.c

```diff
@@ -210,17 +210,21 @@ void __gr_del_entry (const struct commonio_entry *ent)
 
 static int gr_cmp (const void *p1, const void *p2)
 {
+	const struct group *g1, *g2;
 	gid_t u1, u2;
 
-	if ((*(struct commonio_entry **) p1)->eptr == NULL) {
+	g1 = (*(const struct commonio_entry *const *) p1)->eptr;
```

Comment:
How about using another temporary variable to avoid this cast?  It's rather confusing.

Something like this?

```c
const struct commonio_entry *const *ce1;
const struct commonio_entry *const *ce2;

[...]

ce1 = p1;
g1 = (*ce1)->eptr;

[...]

ce2 = p2;
g2 = (*ce2)->eptr;
```

(I wrote the two definitions in separate lines because when there are multiple pointer levels and the pointers have qualifiers, it gets quite confusing to have multiple ones in one line.)

Reply:
seems reasonable, adopted your suggestion